unity/ARUI/Assets/Scripts/hl2ss.cs

```diff
@@ -7,6 +7,22 @@
 
 public class hl2ss : MonoBehaviour
 {
+    private Logger _logger = null;
+
+    /// <summary>
+    /// Lazy acquire the ARUI logger object and return the reference to it.
+    /// </summary>
+    /// <returns>Logger instance reference.</returns>
+    private ref Logger logger()
```

Comment:
Does this want/need any thread-safe protections? I'm assuming you tested this and didn't have any issues, so I may be prematurely concerned.

Reply:
Yeah, I tested this and it seemed OK. We only log the IP address at the beginning, so hopefully there aren't any threading issues there.